src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts

```diff
@@ -122,17 +219,19 @@ export function useSwapInputsController({
     }
 
     return niceIncrementFormatter(
-      incrementDecimalPlaces,
-      Number(assetToSell.value?.balance.amount),
-      Number(assetToSell.value?.native.price?.amount),
-      niceIncrement,
+      incrementDecimalPlaces.value,
+      balance,
+      price,
+      niceIncrement.value,
       percentageToSwap.value,
       sliderXPosition.value
     );
   });
 
   const formattedInputNativeValue = useDerivedValue(() => {
-    if (inputMethod.value === 'slider' && percentageToSwap.value === 0 && !inputValues.value.inputNativeValue) return '$0.00';
+    if ((inputMethod.value === 'slider' && percentageToSwap.value === 0) || !inputValues.value.inputNativeValue) {
+      return '$0.00';
```

Comment:
native currency shiz

Reply:
related: https://linear.app/rainbow/issue/APP-1346/switch-tolocalestring-to-support-other-nativecurrencies